src/gwt/src/org/rstudio/core/client/hyperlink/WebHyperlink.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * HelpHyperlink.java
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+package org.rstudio.core.client.hyperlink;
+
+import com.google.gwt.user.client.ui.Label;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import com.google.gwt.user.client.ui.Widget;
+
+import org.rstudio.studio.client.RStudioGinjector;
+import org.rstudio.studio.client.server.ServerError;
+import org.rstudio.studio.client.server.ServerRequestCallback;
+import org.rstudio.studio.client.workbench.views.source.model.SourceServerOperations;
+
+public class WebHyperlink extends Hyperlink {
+
+    public WebHyperlink(String url, String params, String text, String clazz) 
+    {
+        super(url, params, text, clazz);
+        server_ = RStudioGinjector.INSTANCE.getServer();
+    }
+
+    @Override
+    public void onClick() 
+    {
+        String code = "utils::browseURL('" + url + "')";
```

Comment:
Does this handle `url`s that haven't been properly URL encoded? Or, do we need to do any other escaping -- e.g. is there a change of someone "browsing" to `https://www.google.com");doBadThings();c("`?

Reply:
Thjis should probably be replaced by something that does not need to go to the R side.